### ISSUE TITLE
React does not recognize the `suffixCls` prop on a DOM element - 5.2.3

### ISSUE THREAD

**Issue by Handpear** [ant-design/ant-design#40968]
### Reproduction link

[https://ant.design/components/layout-cn#layout](https://ant.design/components/layout-cn#layout)

### Steps to reproduce

基本结构，点击codesandbox打开

### What is expected?

无相关报错

### What is actually happening?

Warning: React does not recognize the `suffixCls` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `suffixcls` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

| Environment | Info |
| --- | --- |
| antd | 5.2.3 |
| React | last |
| System | win11 |
| Browser | chrome |

---
应该是由以下更新造成，5.2.2正常
修复 Layout.Header 单独使用时，Layout.colorBgHeader token 配置不生效的问题。#40933

**Comment by DercilioFontes** [comment]
Hi,

I am still getting this warning in v5.2.3 using the Layout component.

<img width="1216" alt="Screenshot 2023-02-28 at 9 22 09 AM" src="https://user-images.githubusercontent.com/20172847/221882580-f1ef498e-7bfe-4ebe-88c6-629650554311.png">

**Comment by yoyo837** [comment]
> Hi,
> 
> I am still getting this warning in v5.2.3 using the Layout component.
> 
> <img alt="Screenshot 2023-02-28 at 9 22 09 AM" width="1216" src="https://user-images.githubusercontent.com/20172847/221882580-f1ef498e-7bfe-4ebe-88c6-629650554311.png">

Pls waiting next release!

**Comment by DercilioFontes** [comment]
Ok. It's true. Sorry about that. My bad.

**Comment by zju1** [comment]
Hello. I'm facing that issue too. Is there any temporary alternative option to fix it?

**Comment by yoyo837** [comment]
> Hello. I'm facing that issue too. Is there any temporary alternative option to fix it?

It should not affect the normal working of the components except for one annoying warning, we will fix it in the next release.

**Comment by netcore-jroger** [comment]
same issue. v5.2.3

**Comment by nodejavascript** [comment]
Shouldn't this issue be reopened?

**Comment by afc163** [comment]
<img width="864" alt="图片" src="https://user-images.githubusercontent.com/507615/222474294-fa270941-242e-4d15-b9a3-74cd0c3feaca.png">

**Comment by nodejavascript** [comment]
@afc163 Im still getting this issue in v5.2.3.


Warning: React does not recognize the `suffixCls` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `suffixcls` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

Bug exists somewhere here:
```
const BasicLayout = /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__.forwardRef((props, ref) => {
  const {
    direction
  } = react__WEBPACK_IMPORTED_MODULE_2__.useContext(_config_provider__WEBPACK_IMPORTED_MODULE_3__.ConfigContext);
  const [siders, setSiders] = react__WEBPACK_IMPORTED_MODULE_2__.useState([]);
  const {
      prefixCls: customizePrefixCls,
      className,
      rootClassName,
      children,
      hasSider,
      tagName: Tag
    } = props,
    others = __rest(props, ["prefixCls", "className", "rootClassName", "children", "hasSider", "tagName"]);
  const {
    getPrefixCls
  } = react__WEBPACK_IMPORTED_MODULE_2__.useContext(_config_provider__WEBPACK_IMPORTED_MODULE_3__.ConfigContext);
  const prefixCls = getPrefixCls('layout', customizePrefixCls);
  const [wrapSSR, hashId] = (0,_style__WEBPACK_IMPORTED_MODULE_4__["default"])(prefixCls);
  const classString = classnames__WEBPACK_IMPORTED_MODULE_1___default()(prefixCls, {
    [`${prefixCls}-has-sider`]: typeof hasSider === 'boolean' ? hasSider : siders.length > 0,
    [`${prefixCls}-rtl`]: direction === 'rtl'
  }, className, rootClassName, hashId);
  const contextValue = react__WEBPACK_IMPORTED_MODULE_2__.useMemo(() => ({
    siderHook: {
      addSider: id => {
        setSiders(prev => [].concat((0,_babel_runtime_helpers_esm_toConsumableArray__WEBPACK_IMPORTED_MODULE_0__["default"])(prev), [id]));
      },
      removeSider: id => {
        setSiders(prev => prev.filter(currentId => currentId !== id));
      }
    }
  }), []);
  return wrapSSR( /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__.createElement(LayoutContext.Provider, {
    value: contextValue
  }, /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__.createElement(Tag, Object.assign({
    ref: ref,
    className: classString
  }, others), children)));
});
```

Im using 
```
"antd": "5.2.3",
"react": "^18.2.0",
"react-dom": "^18.2.0",
```

**Comment by DercilioFontes** [comment]
Guys! The title of the issue is reporting that it is in v5.2.3. It was confusing for me. It thought it was fixed in v5.2.3. It was closed because it was already merged a fix and it is for the next release v5.2.4.
